payments-core/src/main/java/com/stripe/android/payments/core/authentication/threeds2/Stripe3ds2ChallengeResultProcessor.kt

```diff
@@ -0,0 +1,228 @@
+package com.stripe.android.payments.core.authentication.threeds2
+
+import com.stripe.android.Logger
+import com.stripe.android.StripeIntentResult
+import com.stripe.android.exception.StripeException
+import com.stripe.android.networking.AnalyticsEvent
+import com.stripe.android.networking.AnalyticsRequestExecutor
+import com.stripe.android.networking.AnalyticsRequestFactory
+import com.stripe.android.networking.ApiRequest
+import com.stripe.android.networking.RetryDelaySupplier
+import com.stripe.android.networking.StripeRepository
+import com.stripe.android.payments.PaymentFlowResult
+import com.stripe.android.payments.core.injection.ENABLE_LOGGING
+import com.stripe.android.stripe3ds2.transaction.ChallengeResult
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.withContext
+import javax.inject.Inject
+import javax.inject.Named
+import javax.inject.Singleton
+import kotlin.coroutines.CoroutineContext
+
+internal interface Stripe3ds2ChallengeResultProcessor {
+    suspend fun process(challengeResult: ChallengeResult): PaymentFlowResult.Unvalidated
+}
+
+@Singleton
+internal class DefaultStripe3ds2ChallengeResultProcessor @Inject constructor(
+    private val stripeRepository: StripeRepository,
+    private val analyticsRequestExecutor: AnalyticsRequestExecutor,
+    private val analyticsRequestFactory: AnalyticsRequestFactory,
+    private val retryDelaySupplier: RetryDelaySupplier,
+    @Named(ENABLE_LOGGING) enableLogging: Boolean,
+    private val workContext: CoroutineContext
+) : Stripe3ds2ChallengeResultProcessor {
```

Comment:
@ccen-stripe any guidance on adding this to the object graph? I think I should be able to use a `@Binds` annotation here.

Reply:
Good call, will remove `provideStripe3ds2ChallengeResultProcessor`